torch/_inductor/package/package.py

```diff
@@ -241,9 +248,19 @@ def get_metadata(self) -> Dict[str, str]:
         return self.loader.get_metadata()  # type: ignore[attr-defined]
 
 
-def load_package(path: str, model_name: str = "model") -> AOTICompiledModel:  # type: ignore[type-arg]
-    if not path.endswith(".pt2"):
-        raise RuntimeError("Unable to load package. Path must be a .pt2 file.")
+def load_package(path: Union[str, io.BytesIO], model_name: str = "model") -> AOTICompiledModel:  # type: ignore[type-arg]
+    assert isinstance(path, io.BytesIO) or (
+        isinstance(path, str) and path.endswith(".pt2")
+    ), f"Unable to load package. Path must be a buffer or a file ending in .pt2. Instead got {path}"
+
+    if isinstance(path, io.BytesIO):
+        with tempfile.NamedTemporaryFile(suffix=".pt2") as f:
```

Comment:
just to make sure, this will get deleted after we call the loader, and its okay right?

Reply:
yeah, because we should have an in-memory Runner